flink-state-backends/flink-statebackend-forst/src/test/java/org/apache/flink/state/forst/ForStStateExecutorTest.java

```diff
@@ -0,0 +1,134 @@
+import org.apache.flink.runtime.asyncprocessing.StateRequestContainer;
+import org.apache.flink.runtime.asyncprocessing.StateRequestType;
+import org.apache.flink.runtime.state.KeyGroupRangeAssignment;
+import org.apache.flink.runtime.state.v2.InternalKeyedState;
+
+import org.junit.jupiter.api.Test;
+import org.rocksdb.WriteOptions;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/** Unit test for {@link ForStStateExecutor}. */
+public class ForStStateExecutorTest extends ForStDBOperationTestBase {
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testExecuteValueStateRequest() throws Exception {
+        ForStStateExecutor forStStateExecutor = new ForStStateExecutor(4, db, new WriteOptions());
+        ForStValueState<Integer, String> state1 = buildForStValueState("value-state-1");
+        ForStValueState<Integer, String> state2 = buildForStValueState("value-state-2");
+
+        StateRequestContainer stateRequestContainer =
+                forStStateExecutor.createStateRequestContainer();
+        assertTrue(stateRequestContainer.isEmpty());
+
+        // 1. Update value state: keyRange [0, keyNum)
+        int keyNum = 1000;
+        for (int i = 0; i < keyNum; i++) {
+            ForStValueState<Integer, String> state = (i % 2 == 0 ? state1 : state2);
+            stateRequestContainer.offer(
+                    buildStateRequest(state, StateRequestType.VALUE_UPDATE, i, "test-" + i, i * 2));
+        }
+
+        forStStateExecutor.executeBatchRequests(stateRequestContainer);
+
+        List<StateRequest<?, ?, ?>> checkList = new ArrayList<>();
+        stateRequestContainer = forStStateExecutor.createStateRequestContainer();
+        // 2. Get value state: keyRange [0, keyNum)
+        //    Update value state: keyRange [keyNum, keyNum + 100]
+        for (int i = 0; i < keyNum; i++) {
+            ForStValueState<Integer, String> state = (i % 2 == 0 ? state1 : state2);
+            StateRequest<?, ?, ?> getRequest =
+                    buildStateRequest(state, StateRequestType.VALUE_GET, i, null, i * 2);
+            stateRequestContainer.offer(getRequest);
+            checkList.add(getRequest);
+        }
+        for (int i = keyNum; i < keyNum + 100; i++) {
+            ForStValueState<Integer, String> state = (i % 2 == 0 ? state1 : state2);
+            stateRequestContainer.offer(
+                    buildStateRequest(state, StateRequestType.VALUE_UPDATE, i, "test-" + i, i * 2));
+        }
+        forStStateExecutor.executeBatchRequests(stateRequestContainer).get();
+
+        // 3. Check value state Get result : [0, keyNum)
+        for (StateRequest<?, ?, ?> getRequest : checkList) {
+            assertThat(getRequest.getRequestType()).isEqualTo(StateRequestType.VALUE_GET);
+            int key = (int) getRequest.getRecordContext().getKey();
```

Comment:
```suggestion
            int key = (Integer) getRequest.getRecordContext().getKey();
```
otherwise it will fail the compilation.